debug/src/debug.js

```diff
@@ -56,6 +65,44 @@ export function initDebug() {
 			throw new Error('Invalid type passed to createElement(): '+(Array.isArray(type) ? 'array' : type));
 		}
 
+		if ((type==='thead' || type==='tfoot' || type==='thead') && parentType!=='table') {
+			console.error(
+				'Improper nesting of table.' +
+				'Your <thead/tbody/tfoot> should have a <table> parent.'
+				+ serializeVNode(vnode)
+			);
+		}
+		else if (type==='tr' && (parentType!=='thead' && parentType!=='tfoot' && parentType!=='tbody')) {
+			console.error(
+				'Improper nesting of table.' +
+				'Your <tr> should have a <thead/tbody/tfoot> parent.'
+				+ serializeVNode(vnode)
+			);
+		}
+		else if (type==='td') {
+			if (parentType!=='tr') {
+				console.error(
+					'Improper nesting of table.' +
+					'Your <td> should have a <tr> parent.'
+					+ serializeVNode(vnode)
+				);
+			}
+			else if (parent._parent.type === 'thead') {
```

Comment:
This will break when the `td` is rendered by a component. We likely need to do another call to `getClosestDomParent`:

```jsx
function Cell() {
  return <td>foo</td>;
}

<thead>
  <Cell />
</thead>
```